plugin/storage/cassandra/factory.go

```diff
@@ -147,6 +152,19 @@ func (f *Factory) CreateArchiveSpanWriter() (spanstore.Writer, error) {
 	return cSpanStore.NewSpanWriter(f.archiveSession, f.Options.SpanStoreWriteCacheTTL, f.archiveMetricsFactory, f.logger, options...), nil
 }
 
+// CreateLockAndSamplingStore creates a distributedlock.Lock and samplingstore.Store for use with adaptive sampling
+func (f *Factory) CreateLockAndSamplingStore() (distributedlock.Lock, samplingstore.Store, error) {
+	hostname, err := os.Hostname() // todo(jpe): wire up --sampling.override-hostname in ./plugin/sampling/strategystore/adaptive/options.go?
```

Comment:
Now I see why you need the override :-) I think JGroups used to add a short random string to the hostname in order to allow multiple instances on the same hostname.

Reply:
I prefer this approach and will use it. I was having trouble wiring up the `sampling.override-hostname` param to the two places that needed it.